packages/nodes-base/nodes/Merge/v2/MergeV2.node.ts

```diff
@@ -0,0 +1,507 @@
+			}
+		}
+
+		if (mode === 'multiplex') {
+			const clashHandling = this.getNodeParameter(
+				'options.clashHandling.values',
+				0,
+				{},
+			) as IDataObject;
+
+			let input1 = this.getInputData(0);
+			let input2 = this.getInputData(1);
+
+			if (clashHandling.resolveClash === 'preferInput1') {
+				[input1, input2] = [input2, input1];
+			}
+
+			if (clashHandling.resolveClash === 'addSuffix') {
+				input1 = addSuffixToEntriesKeys(input1, '1');
+				input2 = addSuffixToEntriesKeys(input2, '2');
+			}
+
+			const mergeIntoSingleObject = selectMergeMethod(clashHandling);
+
+			if (!input1 || !input2) {
+				return [returnData];
+			}
+
+			let entry1: INodeExecutionData;
+			let entry2: INodeExecutionData;
+
+			for (entry1 of input1) {
+				for (entry2 of input2) {
+					returnData.push({
+						json: {
+							...mergeIntoSingleObject(entry1.json, entry2.json),
+						},
+						binary: {
+							...merge({}, entry1.binary, entry2.binary),
+						},
+						pairedItem: [
+							entry1.pairedItem as IPairedItemData,
+							entry2.pairedItem as IPairedItemData,
+						],
+					});
+				}
+			}
+			return [returnData];
+		}
+
+		if (mode === 'matchPositions') {
+			const clashHandling = this.getNodeParameter(
+				'options.clashHandling.values',
+				0,
+				{},
+			) as IDataObject;
+			const includeUnpaired = this.getNodeParameter('options.includeUnpaired', 0, false) as boolean;
+
+			let input1 = this.getInputData(0);
+			let input2 = this.getInputData(1);
+
+			if (clashHandling.resolveClash === 'preferInput1') {
+				[input1, input2] = [input2, input1];
+			}
+
+			if (clashHandling.resolveClash === 'addSuffix') {
+				input1 = addSuffixToEntriesKeys(input1, '1');
+				input2 = addSuffixToEntriesKeys(input2, '2');
+			}
+
+			if (input1 === undefined || input1.length === 0) {
+				if (includeUnpaired) {
+					return [input2];
+				}
+				return [returnData];
+			}
+
+			if (input2 === undefined || input2.length === 0) {
+				if (includeUnpaired) {
+					return [input1];
+				}
+				return [returnData];
+			}
+
+			let numEntries: number;
+			if (includeUnpaired) {
+				numEntries = Math.max(input1.length, input2.length);
+			} else {
+				numEntries = Math.min(input1.length, input2.length);
+			}
+
+			const mergeIntoSingleObject = selectMergeMethod(clashHandling);
+
+			for (let i = 0; i < numEntries; i++) {
+				if (i >= input1.length) {
+					returnData.push(input2[i]);
+					continue;
+				}
+				if (i >= input2.length) {
+					returnData.push(input1[i]);
+					continue;
+				}
+
+				const entry1 = input1[i];
+				const entry2 = input2[i];
+
+				returnData.push({
+					json: {
+						...mergeIntoSingleObject(entry1.json, entry2.json),
+					},
+					binary: {
+						...merge({}, entry1.binary, entry2.binary),
+					},
+					pairedItem: [entry1.pairedItem as IPairedItemData, entry2.pairedItem as IPairedItemData],
+				});
+			}
+		}
+
+		if (mode === 'matchFields') {
```

Comment:
When matching fields we are only getting the `json` property, causing files to be ignored and lost.
For all other operations such as Multiplex, Match by Position and Append it works fine - binary files work as expected. Only when `Match Fields` is used that binary files are lost.